rt/databinding/jaxb/src/main/java/org/apache/cxf/jaxb/WrapperHelperClassLoader.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.jaxb;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+import org.apache.cxf.Bus;
+import org.apache.cxf.common.spi.GeneratedClassClassLoader;
+import org.apache.cxf.databinding.WrapperHelper;
+
+public class WrapperHelperClassLoader extends GeneratedClassClassLoader implements WrapperHelperCreator {
+    public WrapperHelperClassLoader(Bus bus) {
+        super(bus);
+    }
+
+    @Override
+    public WrapperHelper compile(Class<?> wrapperType, Method[] setMethods, Method[] getMethods,
+                                 Method[] jaxbMethods, Field[] fields, Object objectFactory) {
+
+        int count = 1;
+        String newClassName = wrapperType.getName() + "_WrapperTypeHelper" + count;
+
+        Class<?> cls = loadClass(newClassName, WrapperHelperClassLoader.class);
+        while (cls != null) {
+            try {
+                WrapperHelper helper = WrapperHelper.class.cast(cls.getDeclaredConstructor().newInstance());
+                if (!helper.getSignature().equals(WrapperHelperCompiler.computeSignature(setMethods, getMethods))) {
+                    count++;
+                    newClassName = wrapperType.getName() + "_WrapperTypeHelper" + count;
+                    cls = loadClass(newClassName, WrapperHelperClassLoader.class);
+                } else {
+                    return helper;
+                }
+            } catch (Exception e) {
```

Comment:
Please catch `Throwable` here to cover `XxxError` (like `VerifyError` fe)

Reply:
Strange because I have check previous code and it was the same...

Reply:
Yes, the code seems to be the fine but the context has changed, I believe it corresponds to this block:

```
try {
            return new WrapperHelperCompiler(wrapperType,
                                        setMethods,
                                        getMethods,
                                        jaxbMethods,
                                        fields,
                                        objectFactory).compile();

        } catch (Throwable t) {
            // Some error - probably a bad version of ASM or similar
        }
        return null;
```

Reply:
Ensuring you use COMPUTE_FRAMES and visitMaxs(0, 0) all will be computed but if some class writer don't use that, the refactoring can have changed some stack and the hardcoded ones can be wrong now. One thing to try is to dump the .class, convert it in .java (using jd-gui or so) and try to import it in a project, sometimes something obvious appear.

Reply:
@rmannibucau @reta just implement your 2 feedbacks.